checkmedia.c

```diff
@@ -1,598 +1,205 @@
+  if(media->iso_blocks) {
+    printf(
+      "   iso size: %u%s kB\n",
+      media->iso_blocks >> 1,
+      (media->iso_blocks & 1) ? ".5" : ""
+    );
 
-  if(iso.digest.got_current) {
-    printf("%7s: ", iso.digest.name);
-    for(i = 0; i < iso.digest.size; i++) printf("%02x", iso.digest.full[i]);
-    printf("\n");
+    if(media->pad_blocks) printf("        pad: %u kB\n", media->pad_blocks >> 1);
   }
 
-  return iso.digest.ok ? 0 : 1;
-}
-
-
-/*
- * Calculate digest over iso.
- *
- * Normal digest, except that we assume
- *   - 0x0000 - 0x01ff is filled with zeros (0)
- *   - 0x8373 - 0x8572 is filled with spaces (' ').
- */
-void do_digest(char *file)
-{
-  unsigned char buffer[64 << 10]; /* at least 36k! */
-  int fd, err = 0;
-  unsigned chunks = (iso.size - iso.pad) / (sizeof buffer >> 10);
-  unsigned chunk, u;
-  unsigned last_size = ((iso.size - iso.pad) % (sizeof buffer >> 10)) << 10;
-  
-  if((fd = open(file, O_RDONLY | O_LARGEFILE)) == -1) return;
-
-  printf("  check:     "); fflush(stdout);
-
-  digest_media_init(&iso.digest.ctx);
-  digest_media_init(&iso.digest.full_ctx);
-
-  for(chunk = 0; chunk < chunks; chunk++) {
-    if((u = read(fd, buffer, sizeof buffer)) != sizeof buffer) {
-      err = 1;
-      if(u > sizeof buffer) u = 0 ;
-      iso.err_ofs = (u >> 10) + chunk * (sizeof buffer >> 10);
-      break;
-    };
-
-    digest_media_process(&iso.digest.full_ctx, buffer, sizeof buffer);
-
-    if(chunk == 0) {
-      memset(buffer, 0, 0x200);
-      memset(buffer + 0x8373, ' ', 0x200);
-    }
-
-    digest_media_process(&iso.digest.ctx, buffer, sizeof buffer);
-
-    update_progress((chunk + 1) * (sizeof buffer >> 10));
+  if(media->part_blocks) {
+    printf(
+      "  partition: start %u%s kB, size %u%s kB\n",
+      media->part_start >> 1,
+      (media->part_start & 1) ? ".5" : "",
+      media->part_blocks >> 1,
+      (media->part_blocks & 1) ? ".5" : ""
+    );
   }
 
-  if(!err && last_size) {
-    if((u = read(fd, buffer, last_size)) != last_size) {
-      err = 1;
-      if(u > sizeof buffer) u = 0;
-      iso.err_ofs = (u >> 10) + chunk * (sizeof buffer >> 10);
-    }
-    else {
-      digest_media_process(&iso.digest.ctx, buffer, last_size);
-      digest_media_process(&iso.digest.full_ctx, buffer, last_size);
-
-      update_progress(iso.size - iso.pad);
+  if(opt.verbose) {
+    if(media->full_blocks) {
+      printf(
+        "  full size: %u%s kB\n",
+        media->full_blocks >> 1,
+        (media->full_blocks & 1) ? ".5" : ""
+      );
     }
-  }
 
-  if(!err) {
-    memset(buffer, 0, 2 << 10);		/* 2k */
-    for(u = 0; u < (iso.pad >> 1); u++) {
-      if(!iso.pad_is_skip) digest_media_process(&iso.digest.ctx, buffer, 2 << 10);
-      digest_media_process(&iso.digest.full_ctx, buffer, 2 << 10);
-
-      update_progress(iso.size - iso.pad + ((u + 1) << 1));
+    if(mediacheck_digest_valid(media->digest.iso)) {
+      printf("    iso ref: %s\n", mediacheck_digest_hex_ref(media->digest.iso));
     }
```

Comment:
Just idea: what about write in else in verbose mode that this not iso?

Reply:
I don't what to report all the things it is *not*. The user (well, I, for example) doesn't care.